cmd/cli/docker/docker_run.go

```diff
@@ -139,101 +134,127 @@ func dockerRun(cmd *cobra.Command, cmdArgs []string, opts *DockerRunOptions) err
 
 	image := cmdArgs[0]
 	parameters := cmdArgs[1:]
-	j, err := CreateJob(ctx, image, parameters, opts)
+	job, err := CreateJobDocker(image, parameters, opts)
 	if err != nil {
 		return fmt.Errorf("creating job: %w", err)
 	}
 
-	if err := legacy_job.VerifyJob(ctx, j); err != nil {
-		if _, ok := err.(*bacerrors.ImageNotFound); ok {
-			return fmt.Errorf("docker image '%s' not found in the registry, or needs authorization", image)
-		} else {
-			return fmt.Errorf("verifying job: %s", err)
-		}
-	}
-
-	quiet := opts.RunTimeSettings.PrintJobIDOnly
-	if !quiet {
-		containsTag := dockerImageContainsTag(image)
-		if !containsTag {
-			cmd.PrintErrln("Using default tag: latest. Please specify a tag/digest for better reproducibility.")
-		}
+	// Normalize and validate the job spec
+	job.Normalize()
+	if err := job.ValidateSubmission(); err != nil {
+		return fmt.Errorf("%s: %w", userstrings.JobSpecBad, err)
 	}
 
+	// TODO(forrest) [refactor]: this options is _almost_ useful. At present it marshals the entire
+	// job spec to yaml, said spec cannot be used with `bacalhau job run` since it contains fields that
+	// users are not permitted to set, like ID, Version, ModifyTime, State, etc.
+	// The solution here is to have a "JobSubmission" type that is different from the actual job spec.
```

Comment:
Looking for opinions on this. We don't need to land the proposed change as part of this PR, but it would be good to do soon

Reply:
Generally speaking, intermediate representation are not necessarily a bad thing. Forcing the user or user layer to understand the capabilities of what is required at the submission level is not going to give a great UX.

However, this is going to be an area we end up coming back to often, because this translation will change when we decide to change internal functions.

Reply:
Is this for or against a separate type for job submissions? We could even embed said type in the current Job model.

Reply:
Conversation here related to this: https://github.com/bacalhau-project/bacalhau/issues/3719#issuecomment-2061238679

Reply:
I hear your point. My preference is to delay any decision to change our API specs and have different ways to submit jobs until we collect more feedback about current APIs, and have more dedicated time to explore alternatives and align on the right option. Options might include splitting the job type into spec and state top level fields, where state hold the system defined fields, and spec is what the user submits. Again, I truly don't recommend figuring this out just yet.

A low hanging fruit is to introduce SanitizeForSubmission() method under models.Job that will clear out all fields that are reserved by the system so that something like bacalhau job describe <job_id> --output yaml | bacalhau job run would work

Reply:
The "current APIs" are the ones serving `bacalhau create/list/describe/etc.` - AKA the V1 APIs.
The APIs we are moving to in this PR are still considered "experimental" - AKA the V2 APIs. They are not the canonical bacalhau APIs we expected user to be interacting with. We should be striving to get them right before moving to them. 

> My preference is to delay any decision to change our API specs and have different ways to submit jobs until we collect more feedback about current APIs, and have more dedicated time to explore alternatives and align on the right option.

Given the above, I strongly disagree with this point. We shouldn't be putting off improvements to the new APIs we are moving to. The whole point of creating these new API was to make improvements to it before they become canonical. As you have already alluded to, and outlined in greater detail via https://github.com/bacalhau-project/bacalhau/issues/3983, separating the system representation of a job from the user submission is an excellent improvement - both from a development perspective and user perspective.
We have made it a habit of encouraging the development team to make improvements to the V2 APIs before we make them the default, let's ensure we allow that before moving forward.

> Again, I truly don't recommend figuring this out just yet.

Just the opposite: I would recommend we figure this out now, since if we don't, we'll have to introduce yet another API - V3 - as changing the payload and responses expected by the V2 API will be a breaking change. 

> A low hanging fruit is to introduce SanitizeForSubmission() method under models.Job that will clear out all fields that are reserved by the system so that something like bacalhau job describe <job_id> --output yaml | bacalhau job run would work

I believe this confuses low hanging fruit "accept technical debt", and as I've stated above, given these are still experimental APIs we should be making the effort to implement them correctly, rather than postpone valuable improvements. 

> have more dedicated time to explore alternatives and align on the right option

The change here is easier than we think. Initially we can keep it small by simply changing the request payload expected by the APIs to a subset of the Job spec as described in https://github.com/bacalhau-project/bacalhau/issues/3983. We don't need to fully separate the spec from the state across the system as a whole until later. And said change can be made without affecting the API schema later.

Reply:
This was discussed somewhere else and agreed not to change things for now